### PR TITLE
Scan through members in `InitializationFieldAccessVisitor`

### LINKED ISSUE #612: NullPointerException in GenericAnnotatedTypeFactory.getFirstNodeOfKindForTree(
New in 3.39.0-eisop1:

```java
interface A {}

@interface B {
  Class<? extends A> value();
}

class C implements A {}

class E {}

public @interface Z {
  abstract class D implements A {

    public D() {
      g(new E());
    }

    static void g(Object... xs) {}
  }
}
```

```
error: SourceChecker.typeProcess: unexpected Throwable (NullPointerException) while processing Z.java
  ; The Checker Framework crashed.  Please report the crash.  Version: Checker Framework 3.39.0-eisop1.
  Checker: class org.checkerframework.checker.nullness.NullnessChecker
  Visitor: class org.checkerframework.checker.initialization.InitializationVisitor
  Compilation unit: Z.java
  Last visited tree at line 15 column 13:
        g(new E());
  Exception: java.lang.NullPointerException; java.lang.NullPointerException
        at org.checkerframework.framework.type.GenericAnnotatedTypeFactory.getFirstNodeOfKindForTree(GenericAnnotatedTypeFactory.java:1299)
        at org.checkerframework.framework.type.GenericAnnotatedTypeFactory.getAnnotatedTypeVarargsArray(GenericAnnotatedTypeFactory.java:1839)
        at org.checkerframework.common.basetype.BaseTypeVisitor.checkVarargs(BaseTypeVisitor.java:2129)
        at org.checkerframework.common.basetype.BaseTypeVisitor.visitMethodInvocation(BaseTypeVisitor.java:2012)
        at org.checkerframework.checker.initialization.InitializationVisitor.visitMethodInvocation(InitializationVisitor.java:299)
        at org.checkerframework.checker.initialization.InitializationVisitor.visitMethodInvocation(InitializationVisitor.java:66)
        at jdk.compiler/com.sun.tools.javac.tree.JCTree$JCMethodInvocation.accept(JCTree.java:1666)
        at jdk.compiler/com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:82)
        at org.checkerframework.framework.source.SourceVisitor.scan(SourceVisitor.java:95)
        at org.checkerframework.common.basetype.BaseTypeVisitor.scan(BaseTypeVisitor.java:437)
        at org.checkerframework.common.basetype.BaseTypeVisitor.scan(BaseTypeVisitor.java:194)
        at jdk.compiler/com.sun.source.util.TreeScanner.visitExpressionStatement(TreeScanner.java:433)
        at jdk.compiler/com.sun.tools.javac.tree.JCTree$JCExpressionStatement.accept(JCTree.java:1476)
        at jdk.compiler/com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:82)
        at org.checkerframework.framework.source.SourceVisitor.scan(SourceVisitor.java:95)
        at org.checkerframework.common.basetype.BaseTypeVisitor.scan(BaseTypeVisitor.java:437)
        at org.checkerframework.common.basetype.BaseTypeVisitor.scan(BaseTypeVisitor.java:194)
        at jdk.compiler/com.sun.source.util.TreeScanner.scanAndReduce(TreeScanner.java:90)
        at jdk.compiler/com.sun.source.util.TreeScanner.scan(TreeScanner.java:105)
        at jdk.compiler/com.sun.source.util.TreeScanner.visitBlock(TreeScanner.java:248)
        at jdk.compiler/com.sun.tools.javac.tree.JCTree$JCBlock.accept(JCTree.java:1048)
        at jdk.compiler/com.sun.source.util.TreePathScanner.scan(TreePathScanner.java:82)
```

### PR DESCRIPTION
**Pull request by wmdietl**
The problem only occurs when the enclosing type is an annotation type, e.g. if `Issue612Min` is changed from an `@interface` to a `class`, the NPE doesn't happen.
So maybe there is some more localized fix that undoes some special handling for annotation types.

@flo2702 can you have a look at this and see whether you find a better fix? The issue was introduced with #444 .

Fixes #612 .

### CODE REVIEW

File: checker/src/main/java/org/checkerframework/checker/initialization/InitializationFieldAccessVisitor.java [hunk 1]
```diff
@@ -24,6 +28,9 @@ public void processClassTree(ClassTree classTree) {
         // and InitializationChecker, this checker performs the flow analysis
         // (which is handled in the BaseTypeVisitor), but does not perform
         // any type checking.
-        // Thus, this method does nothing.
+        // Thus, this method does nothing but scan through the members.
+        if (!assumeInitialized) {
```
Comment (wmdietl) [inline checker/src/main/java/org/checkerframework/checker/initialization/InitializationFieldAccessVisitor.java:32]: @flo2702 This is the key change in this PR. Sorry for adding so many other changes while tracking this down.
I think I'll merge this PR now, to fix the exception. Can you open a new PR if you find a better fix?

File: checker/tests/nullness-extra/issue5174/Issue5174.out [hunk 2]
```diff
@@ -413,12 +413,12 @@ Before: InitializationStore#93(
 ~~~~~~~~~
 o   [ VariableDeclaration ]
 (this)   [ ImplicitThis ]
-(this).f   [ FieldAccess ]    > CFAV{@Initialized, Object}
-o = (this).f   [ Assignment ]    > CFAV{@Initialized, Object}
+(this).f   [ FieldAccess ]    > CFAV{, T}
```
Comment (wmdietl) [inline checker/tests/nullness-extra/issue5174/Issue5174.out:416]: @flo2702 These changes come a bit as a surprise, but look good to me - field `f` has type `T`, so the new types look more precise than before, where it used the erasure.

File: checker/src/main/java/org/checkerframework/checker/initialization/InitializationFieldAccessVisitor.java (hunk 1 above)
Comment (flo2702) [reply checker/src/main/java/org/checkerframework/checker/initialization/InitializationFieldAccessVisitor.java:32]: Agreed. I'll keep looking, but haven't found anything better.

File: checker/tests/nullness-extra/issue5174/Issue5174.out (hunk 2 above)
Comment (flo2702) [reply checker/tests/nullness-extra/issue5174/Issue5174.out:416]: Seems like scanning through all members once in the visitor somehow allows more precise types to be used during the dataflow refinement. Probably something to consider while I look for a better fix, but I admit that I don't really understand how the visitor affects the refinement in this way.

**Pull request merged**